### PR TITLE
[3.x] C#: Print error when MethodBind call fails

### PR DESCRIPTION
**Pull request by raulsntos**
- `3.x` version of https://github.com/godotengine/godot/pull/79249.
- Fixes https://github.com/godotengine/godot/issues/46645.

### CODE REVIEW

File: modules/mono/mono_gd/gd_mono_internals.cpp [hunk 1]
```diff
@@ -139,4 +139,89 @@ void gd_unhandled_exception_event(MonoException *p_exc) {
...
+		}
+
+	} else {
+		basestr = Variant::get_type_name(p_var->get_type());
+	}
+
+	return basestr;
+}
+
+static String _get_call_where(const String &p_method, const Variant *p_instance, const Variant **argptrs, int argc) {
```
Comment (RedworkDE) [inline modules/mono/mono_gd/gd_mono_internals.cpp:170]: This logic was adjusted in the 4.x version, but I suppose we could also just keep this closer to the GDScript version here as it is basically entirely different code here anyways.

**Review by RedworkDE — COMMENTED** [review]
Code changes look sensible to me, but I don't actually have a setup to build and test the 3.x mono version.

File: modules/mono/mono_gd/gd_mono_internals.cpp (hunk 1 above)
Comment (raulsntos) [reply modules/mono/mono_gd/gd_mono_internals.cpp:170]: I think it's safer to keep the code as close as the GDScript version for 3.x.

**Review by RedworkDE — APPROVED** [review]
Needs a 3.x version of #79342 (Now exists as https://github.com/godotengine/godot/pull/80072) to open the source files correctly on windows, works as expected otherwise.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**